src/adagrams.js

```diff
@@ -1,8 +1,150 @@
+const letterPool = {
+  "A": 9,
+  "B": 2,
+  "C": 2,
+  "D": 4,
+  "E": 12,
+  "F": 2,
+  "G": 3,
+  "H": 2,
+  "I": 9,
+  "J": 1,
+  "K": 1,
+  "L": 4,
+  "M": 2,
+  "N": 6,
+  "O": 8,
+  "P": 2,
+  "Q": 1,
+  "R": 6,
+  "S": 4,
+  "T": 6,
+  "U": 4,
+  "V": 2,
+  "W": 2,
+  "X": 1,
+  "Y": 2,
+  "Z": 1
+};
+
+const letterScores = {
+  "A": 1,
+  "B": 3,
+  "C": 3,
+  "D": 2,
+  "E": 1,
+  "F": 4,
+  "G": 2,
+  "H": 4,
+  "I": 1,
+  "J": 8,
+  "K": 5,
+  "L": 1,
+  "M": 3,
+  "N": 1,
+  "O": 1,
+  "P": 3,
+  "Q": 10,
+  "R": 1,
+  "S": 1,
+  "T": 1,
+  "U": 1,
+  "V": 4,
+  "W": 4,
+  "X": 8,
+  "Y": 4,
+  "Z": 10
+};
 const Adagrams = {
+  countItems(collection, letter) {
+    let count = 0;
+    for(let i = 0; i < collection.length; i += 1) {
+      if (collection[i] === letter) {
+        count += 1;
+      }
+    }
+    
+    return count;
+  },
+
   drawLetters() {
-    // Implement this method for wave 1
+    let letterDraw = [];
+    Object.keys(letterPool).forEach((letter) => {
+      let i = letterPool[letter];
+  
+      while (i > 0) {
+        letterDraw.push(letter)
+        i -= 1
+      }
+    })
+
+    let hand = [];
+    for (let i = 10; i > 0; i -=1) {
+      let randomLetter = letterDraw[Math.floor(Math.random()*letterDraw.length)];
+
+      while (this.countItems(hand, randomLetter) == letterPool[randomLetter]) {
```

Comment:
This doesn't really solve the problem as proposed. Your logic does prevent more than the max quantity of a letter from being included in the hand, but the odds of getting any letter is 1/26.